test/queryRunner.test.ts

```diff
@@ -727,6 +727,153 @@ suite('Query Runner tests', () => {
+                executionStart: new Date().toISOString(),
+                executionEnd: new Date().toISOString()
+            }]
+        };
+
+        setup(() => {
+            testSqlToolsServerClient.setup(x => x.sendRequest(TypeMoq.It.isAny(),
+                                                            TypeMoq.It.isAny())).callback(() => {
+                                                                // testing
+                                                            }).returns(() => { return Promise.resolve(testresult); });
+            testStatusView.setup(x => x.executingQuery(TypeMoq.It.isAnyString()));
+            testStatusView.setup(x => x.executedQuery(TypeMoq.It.isAnyString()));
+            testVscodeWrapper.setup( x => x.logToOutputChannel(TypeMoq.It.isAnyString()));
+            testVscodeWrapper.setup(x => x.clipboardWriteText(TypeMoq.It.isAnyString())).callback(() => {
+                                                                // testing
+                                                            }).returns(() => { return Promise.resolve(); });
+
+        });
+
+        // ------ Copy tests with multiple selections  -------
+        test('Correctly copy pastes a selection', (done) => {
+            let configResult: {[key: string]: any} = {};
+            configResult[Constants.copyIncludeHeaders] = false;
+            setupWorkspaceConfig(configResult);
+
+            let queryRunner = new QueryRunner(
+                testuri,
+                testuri,
+                testStatusView.object,
+                testSqlToolsServerClient.object,
+                testQueryNotificationHandler.object,
+                testVscodeWrapper.object
+            );
+            queryRunner.uri = testuri;
+            queryRunner.copyResults(testRange, 0, 0).then(() => {
+                testVscodeWrapper.verify<void>(x => x.clipboardWriteText(TypeMoq.It.isAnyString()), TypeMoq.Times.once());
+                done();
+            });
+        });
+
+        test('Copies selection with column headers set in user config', () => {
+            // Set column headers in the user config settings
+            let configResult: {[key: string]: any} = {};
+            configResult[Constants.copyIncludeHeaders] = true;
+            setupWorkspaceConfig(configResult);
+
+            let queryRunner = new QueryRunner(
+                testuri,
+                testuri,
+                testStatusView.object,
+                testSqlToolsServerClient.object,
+                testQueryNotificationHandler.object,
+                testVscodeWrapper.object
+            );
+            queryRunner.uri = testuri;
+            // Call handleResult to ensure column header info is seeded
+            queryRunner.handleQueryComplete(result);
+            return queryRunner.copyResults(testRange, 0, 0).then(() => {
+                testVscodeWrapper.verify<void>(x => x.clipboardWriteText(TypeMoq.It.isAnyString()), TypeMoq.Times.once());
```

Comment:
You never actually check whether the headers were copied or not so this test and the ones below don't seem super useful as is